### DISCUSSION TITLE
Beginner: Help with Trajectory

### DISCUSSION THREAD

**Discussion by liu1223wenxi** [bitcraze/discussions#946] [Q&A]
Hi 

I am reaching out regarding an issue I'm encountering with the Crazyflie 2.1 drone. Specifically, while trying the mocap_hl_commander example, my drone keeps flying into the wall (same side). I've ensured that the trajectory points are well within the constraints of my mocap system.

Firstly: I would appreciate any insights on potential reasons behind this behavior.
Secondly: could you guide me on how to perform position control for a singular point?
Lastly: I am keen on understanding the process to design a custom trajectory. Any resources to help?

**Comment by knmcguire** [comment]
Hi! let me try out here. 

1. You need to first know if your script is getting the right position since it sounds like it is not. There were some discussions on the label 'positioning: mocap' on this so see you can find an answer there, like this one: https://github.com/orgs/bitcraze/discussions/880. 
2. There are several ways to do this. Here is some documentation on how to use the commander structure to send setpoints: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/functional-areas/sensor-to-control/commanders_setpoints/. There are also some examples you can try out in the cflib autonomy folders in examples: https://github.com/bitcraze/crazyflie-lib-python
3. The custom trajectory that you see we generated with a script from @whoenig. https://github.com/whoenig/uav_trajectories
